File: Python_백준/1260_DFS와 BFS.py
#1260
import queue

#dfs

N, M, V = map(int, input().split)

G = [[] for _ in range(N + 1)]

for i in range(M):
    v, u = map(int, input().split())
    G[v].append(u) # v->u
    G[u].append(v) # u->v

for v in range(1, N + 1):
    G[v].sort()

visit = [False for _ in range(N + 1)] #방문횟수

def dfs(v):
    visit[v] = True
    print(v, end="")
    for u in G(v):
        if not visit[u]:
            dfs(u)

dfs(v)
print()

#bfs
visit = [False for _ in range(N + 1)] #방문횟수

q = queue.Queue()
q.put(v)
while not q.empty():
    v = q.get()

    if visit[v]:
        continue

    visit[v] = True
    print(v, end="")

    for u in G(v):
        if not visit[u]:
            q.put(u)

File: Cpp_백준/1260_DFS와 BFS.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>
using namespace std;

vector<int> *graph;
int *visit;
queue<int> q;

void dfs(int node){
    cout << node << " ";
    visit[node] = 1;
    for (int i = 0; i < graph[node].size(); i++){
        if (!visit[graph[node].at(i)]) dfs(graph[node].at(i));
    }
}

void bfs(int node){
    int t;
    q.push(node);
    visit[node] = 1;
    while (!q.empty()){
        t = q.front();
        cout << t << " ";
        for (int i = 0; i < graph[t].size(); i++){
            if (!visit[graph[t].at(i)]){
                visit[graph[t].at(i)] = 1;
                q.push(graph[t].at(i));
            }
        }
        q.pop();
    }
}

int main(){
    int n, m, v; // n : 정점의 개수, m : 간선의 개수, v : 시작 정점
    int tmp1, tmp2;
    cin >> n >> m >> v;

    graph = new vector<int>[n + 1];
    visit = new int[n + 1];

    while(m--){
        cin >> tmp1 >> tmp2;
        graph[tmp1].push_back(tmp2);
        graph[tmp2].push_back(tmp1); //양방향이기 때문에
    }

    for (int i = 1; i <= n; i++) {
        visit[i] = 0;
        sort(graph[i].begin(), graph[i].end());
    }
    dfs(v);
    cout << endl;
    for (int i = 1; i <= n; i++) visit[i] = 0;
    bfs(v);
    cout << endl;
}